### PR TITLE
Minor refactor to reduce variable_force stack use

### PR DESCRIPTION
**Pull request by jlfaber**
* replace dispatch_node with an inline code that uses a lookup table
* change throw/catch to a simple return value check for skipping child node processing

The throw/catch change removes two stack frames for each level of recursive nesting required
when traversing the AST.  The dispatch_node change removes an additional stack frame for levels
where the children are processed by the handler code.

Together these changes should help _a little_ with the general problem described in #5341, in which
ruby underreports the amount of available stack space.

This PR should cause no user-observable behavior changes.

### CODE REVIEW

File: lib/rubocop/cop/variable_force.rb [hunk 1]
```diff
@@ -82,10 +103,8 @@ def investigate(processed_source)
       end
 
       def process_node(node)
-        catch(:skip_children) do
-          dispatch_node(node)
-          process_children(node)
-        end
+        retval = send HANDLERS[node.type], node if HANDLERS.key? node.type
+        process_children(node) unless retval == :SKIP_CHILDREN
```
Comment (bbatsov) [inline lib/rubocop/cop/variable_force.rb]: Why is `:SKIP_CHILDREN` uppercase?

File: lib/rubocop/cop/variable_force.rb [hunk 2]
```diff
@@ -55,6 +55,27 @@ class VariableForce < Force # rubocop:disable Metrics/ClassLength
 
       SEND_TYPE = :send
 
+      handlers = {}
+
+      handlers[VARIABLE_ASSIGNMENT_TYPE] = :process_variable_assignment
```
Comment (bbatsov) [inline lib/rubocop/cop/variable_force.rb]: I wonder if this huge block of init code shouldn't be moved to some class method.

File: lib/rubocop/cop/variable_force.rb [hunk 3]
```diff
@@ -82,10 +103,8 @@ def investigate(processed_source)
       end
 
       def process_node(node)
-        catch(:skip_children) do
-          dispatch_node(node)
-          process_children(node)
-        end
+        retval = send HANDLERS[node.type], node if HANDLERS.key? node.type
```
Comment (bbatsov) [inline lib/rubocop/cop/variable_force.rb]: Please, add parens around the `send` and `key?` params.

File: lib/rubocop/cop/variable_force.rb [hunk 4]
```diff
@@ -55,6 +55,27 @@ class VariableForce < Force # rubocop:disable Metrics/ClassLength
 
       SEND_TYPE = :send
 
+      handlers = {}
```
Comment (bbatsov) [inline lib/rubocop/cop/variable_force.rb]: I'm on the fence about whether the name `handlers` isn't a bit too generic, but I also don't have a strong preference about how to change it.

**Comment by jlfaber** [comment]
Thanks, @bbatsov, for the feedback! I went back to determining the node handler in a method (as it was before), but now instead of calling it immediately, I return the method name so that we can call it directly from `process_node` to save a stack frame.

**Comment by jnardone** [comment]
@bbatsov   given that the upstream ruby fix necessary to really fix the stackdepth calc is stalled, can we at least get this merged in the meantime?

**Comment by bbatsov** [comment]
Yeah, I'm OK with the current version of the code.

**Pull request merged**